resources/uco_design_document.md

```diff
@@ -71,24 +71,56 @@ To achieve its targeted objectives UCO has chosen to pursue an ontological appro
 An RDF graph of this would look like:
 ![RDF Graph example](./object-example-rdf-graph.drawio.png "RDF graph example")
 
-In the UCO RDFS/OWL/SHACL ontology, classes are defined for any relevant domain concept as well as for any structured concept characterizing some aspect of a domain concept. These are structured concept classes that specify into `UcoObject` classes, `Facet` classes and other classes. `UcoObject` and `Facet` classes therefore are structured concept classes, however, `UcoObject` classes and `Facet` classes are disjoint from each other. Moreover, `Facet` classes inhere in `UcoObject` classes; this implies that for a facet concept to exist, it is dependent on the existence of the UcoObject concept that bears the facet. For example, when destroying a red car, the car as bearer for the red color is removed and with it, its red color disappears. Note that the reverse is not true; UcoObjects are not existentially dependent on facets, and, thus, cannot inhere in them. Note further that, although the example suggests that facets are  compulsary for UcoObject concepts, this is not the case.
+In the UCO RDFS/OWL/SHACL ontology, all classes are defined as subclasses of the `UcoThing` class to provide a simple scoping of UCO defined "things".
+Within this scope, classes are defined for any relevant domain concept (domain objects) as well as for any structured concept characterizing some aspect of a domain concept (non-domain objects). To provide clear delineation between these, two disjoint subclasses of `UcoThing` are defined: `UcoObject` (for domain objects) and `UcoInherentCharacterizationThing` (for non-domain objects).
+
+`UcoInherentCharacterizationThing` classes inhere in `UcoObject` classes; this implies that for a `UcoInherentCharacterizationThing` concept to exist, it is dependent on the existence of the `UcoObject` concept that bears the `UcoInherentCharacterizationThing` characterizing it. For example, when destroying a red car, the car as bearer for the red color is removed and with it, its red color disappears. Note that the reverse is not true; `UcoObject` instances are not existentially dependent on `UcoInherentCharacterizationThing` instances, and, thus, cannot inhere in them. Note further that, although the example suggests that `UcoInherentCharacterizationThing` instances are compulsary for `UcoObject` concepts, this is not the case.
+
+Domain concept classes (e.g., File, Action, Identity, Location, Device, etc.) are defined as subclasses of the `UcoObject` class. Non-domain concept classes characterize domain concept classes and are defined as subclasses of the `UcoInherentCharacterizationThing` class. Domain concept classes represent the things whereas non-domain concept classes represent the thing's characteristics. The disjointness between them follows from the fact that the thing can never be the same as its characteristics.
```

Comment:
> Domain concept classes represent the things whereas non-domain concept classes represent the thing's characteristics.  The disjointness between them follows from the fact that the thing can never be the same as its characteristics.

This seems quite odd to me in terms of ontology encoding.

More concretely than me just feeling uneasy: This causes an interoperability challenge with existing ontologies.  Let's say there is some other ontology (the example ontology, prefix `ex:`) that has a property `ex:fileName`, domain `ex:File`.  Further let's say `ex:File` is a "domain concept" within `ex:`.

UCO **can't** define a one-to-one alignment of `observable:fileName` to `ex:fileName` like this:

```turtle
observable:fileName
	owl:equivalentProperty ex:fileName ;
	.
```

This is because the "domain" (in the RDFS sense) of `observable:fileName` is `observable:FileFacet`, a non-domain (in Sean's graph sense) concept.  Note that this `rdfs:domain` is not currently encoded in the ontology, but the spirit of `observable:fileName` is that it is only intended to be on a `Facet` subclass.

Declaring the properties equivalent would make this expression inconsistent with UCO's design:

```turtle
ex:file-1
	a observable:File ;
	ex:fileName "foo.txt" ;
	.
```

OWL expansion would add these statements:

```turtle
# Expand type by recorded property's domain.
ex:file-1 a ex:File .

# Expand property equivalance.
ex:file-1 observable:fileName "foo.txt" .

# Expand type by inferred property's domain.
# (NOTE: This uses an assumed `rdfs:domain` for `observable:fileName`.)
ex:file-1 a observable:FileFacet .

# Inconsistency: Membership of disjoint classes.
ex:file-1
	a
		observable:File ,
		observable:FileFacet
		;
	.
```

It *is* possible to align the properties, but because of this one "hop" through a `Facet`, the alignment must be with a property chain axiom:

```turtle
ex:fileName
	owl:propertyChainAxiom (
		core:hasFacet
		observable:fileName
	) ;
	.
```

Then, expansion from above leads to a consistent output:

```turtle
ex:file-1
	a observable:File ;
	ex:fileName "foo.txt" ;
	.

# Expansion: A temporary node is inferred,
# to satisfy the property chain axiom.
ex:file-1 core:hasFacet $x .
$x observable:fileName "foo.txt" .

# Expansion: Type of _:x inferred from property domain.
$x a observable:FileFacet .
```

(My understanding of the internals of OWL inferencing engines ends about here.  `$x` might be assigned to be a blank node, or it might be discarded after persisting just long enough to see if there are inconsistencies from other inferences.)

The above illustration may be part of why UCO did away with `rdfs:domain` statements on its properties before switching to SHACL.  I've assumed what I think is a more "usual" ontology engineering pattern of using `rdfs:domain`, because UCO currently has *encoded* neither domains, nor top-level properties that could serve to group properties as "Domain-attached," "Inhering," or other terms we could use to say when a property goes on a `UcoObject` vs. a `Facet`.

In the end, I am no closer to understanding the purpose of `Facet`s, or when they should be used to house properties.